server/src/databaseHandlers/surveyDBHandler.ts

```diff
@@ -52,6 +52,9 @@ export default class SurveyDBHandler {
       id: surveyContainerName,
       partitionKey: {
         paths: ['/callId']
+      },
+      uniqueKeyPolicy: {
+        uniqueKeys: [{ paths: ['/callId'] }]
```

Comment:
should the unique key be based on callId and acsUserId?

Reply:
You may need to handle the uniqueKey Constraint exception within your code, in case the duplicate exists. You may be handle the exception and return response as success 200. Some thoughts there on how to handle exception elegantly.

Reply:
If there is a exception throw from `upsert`, the express throw status code 500. Middleware already take care of it.

Reply:
I can put `acsUserId` as unique as well but it will not check `if callId AND acsUserId exist` since it only checks individual's uniqueness

Reply:
The 500 is internal server error which is not clear and too generic for the client side to act upon. Since in this case the server knows the exception is from db and is from unique constraint, it should handle it elegantly and return a status code as may be 409(https://restfulapi.net/http-status-codes/) or just return 200 (as entity is already processed)

Reply:
Since this error will be from `upsert` procedure, Express middleware handles that it returns 500 generic error with message. Let me see what I can do for this.

Reply:
I think that's may be fine within the context that we will anyways show "Thanks for submitting the survey" , so should not matter as we don't need to address for the unique key constraint violation on UI to keep it simple. So I am fine with default 500 error code there as of now

Reply:
Fortunately, I was able to use the route to handle the error more gracefully. I just pushed new changes with test.